.github/workflows/tests.yml

```diff
@@ -0,0 +1,53 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'constructor-manager/**'
+      - 'constructor-manager-cli/**'
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: ${{ matrix.platform }} py${{ matrix.python-version }}
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ['3.8', '3.9', '3.10']
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies cli
+        run: |
+          cd constructor-manager-cli
+          python -m pip install --upgrade pip
+          python -m pip install setuptools tox tox-gh-actions
+          pip list
+
+      # this runs the platform-specific tests declared in tox.ini
+      - name: Test with tox
```

Comment:
Nitpick but update the names here and below as well to indicate which package is being tested.